colossalai/inference/tensor_parallel/engine.py

```diff
@@ -36,17 +57,52 @@ class TPInferEngine:
         >>> generate_kwargs = ...
         >>> shard_config = ShardConfig(enable_tensor_parallelism=True, inference_only=True)
         >>> infer_engine = TPInferEngine(model, shard_config, MAX_BATCH_SIZE, MAX_INPUT_LEN, MAX_OUTPUT_LEN)
+        >>> infer_engine.optimize_model()
```

Comment:
As we discuss in #4577 , the function to optimize the model is not supposed to be exposed to users. What is the purpose of exposing it here?

Reply:
Done